api/src/main/java/org/openmrs/module/coreapps/utils/VisitTypeHelper.java

```diff
@@ -0,0 +1,250 @@
+package org.openmrs.module.coreapps.utils;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.codehaus.jackson.JsonNode;
+import org.codehaus.jackson.JsonParseException;
+import org.codehaus.jackson.map.JsonMappingException;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.codehaus.jackson.node.ArrayNode;
+import org.openmrs.Encounter;
+import org.openmrs.EncounterType;
+import org.openmrs.Location;
+import org.openmrs.Patient;
+import org.openmrs.Person;
+import org.openmrs.Visit;
+import org.openmrs.VisitType;
+import org.openmrs.api.APIException;
+import org.openmrs.api.AdministrationService;
+import org.openmrs.api.EncounterService;
+import org.openmrs.api.VisitService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.coreapps.CoreAppsConstants;
+import org.openmrs.module.emrapi.visit.VisitDomainWrapper;
+import org.springframework.stereotype.Component;
+
+/**
+ * @author Romain Buisson
```

Comment:
Could you take a look at Attribution on this page? https://wiki.openmrs.org/display/docs/Coding+Conventions#CodeStyle